e2e/validators.go

```diff
@@ -1201,3 +1201,64 @@ func ValidateMIGInstancesCreated(ctx context.Context, s *Scenario, migProfile st
 	require.NotContains(s.T, stdout, "No MIG-enabled devices found", "no MIG devices were created.\nOutput:\n%s", stdout)
 	s.T.Logf("MIG instances with profile %s are created", migProfile)
 }
+
+func execOnPodForScenario(ctx context.Context, s *Scenario, pod *corev1.Pod, cmd string) *podExecResult {
+	s.T.Helper()
+	execResult, err := execOnUnprivilegedPod(ctx, s.Runtime.Cluster.Kube, pod.Namespace, pod.Name, cmd)
+	require.NoErrorf(s.T, err, "failed to execute command on pod: %v", cmd)
+	return execResult
+}
+
+// ValidateExecCmdOnVM creates a pod on the target node and executes the given command on that pod.
+func ValidateExecCmdOnVM(ctx context.Context, s *Scenario, command string) {
```

Comment:
don't we already create a testing pod on the e2e nodes? could we just look at adding a readiness/liveness exec probe to that pod? in which case no code changes would even be required